.github/workflows/publish-image.yml

```diff
@@ -0,0 +1,77 @@
+name: Publish Image
+on:
+  release:
+    types:
+      - published
+  push:
+    branches:
+      - main
+  pull_request:
+    branches:
+      - '*'
+
+jobs:
+  build:
+    env:
+      context: .
+      image_name: swoop-caboose
+      image_registry: "${{ secrets.REGISTRY_URI }}"
+      image_repository: "${{ secrets.REGISTRY_REPOSITORY }}"
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@main
+
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v4
+        with:
+          python-version: '3.x'
+
+      - name: Install dependencies
+        run: pip install setuptools-scm
+
+      - name: Get image tags
+        id: image_tags
+        run: |
+          VERSION="$(python -m setuptools_scm | tr + -)"
```

Comment:
I am not sure if using python for this makes the most sense. We can use git to get something comparable without having to install anything else by running `VERSION=$(git describe --tags --abbrev=8 --dirty --always --long)`.

The reason for using `setuptools_scm` on other projects is that it is also the mechanism that we're using for embedding the version information into the python package metadata. golang doesn't have this exactly (it will embed the git commit hash into the binary by default, if git is installed in the build env (another reason to use a multistage build in the Dockerfile, as we probably want to have git installed there). But we can't get a tag into that build metadata for a user-facing version, sadly.

I am looking into options to handle this better (mainly revolving around using the `-ldflags` option of `go build` to set variables within a module that we can use in the code). Maybe I can hand that investigation off to you.

Reply:
Changed ```VERSION=$(git describe --tags --abbrev=8 --dirty --always --long)``` and removed python install